File: RootTools2/src/main/java/com/rarnu/tools/neo/base/BaseLoader.java
package com.rarnu.tools.neo.base;

import android.content.AsyncTaskLoader;
import android.content.Context;

import java.util.List;

public abstract class BaseLoader<T> extends AsyncTaskLoader<List<T>> {

    public BaseLoader(Context context) {
        super(context);
    }

    public abstract List<T> loadInBackground();

    @Override
    protected void onStartLoading() {
        forceLoad();
    }

    @Override
    public void onCanceled(List<T> data) {
        super.onCanceled(data);
    }

    @Override
    protected void onStopLoading() {
        cancelLoad();
    }

    @Override
    protected void onReset() {
        stopLoading();

    }
}

File: RootTools2/src/main/java/com/rarnu/tools/neo/base/BaseLoader.kt
package com.rarnu.tools.neo.base

import android.content.AsyncTaskLoader
import android.content.Context

/**
 * Created by rarnu on 3/23/16.
 */
abstract class BaseLoader<T>(context: Context?) : AsyncTaskLoader<MutableList<T>?>(context) {

    abstract override fun loadInBackground(): MutableList<T>?

    override fun onStartLoading() = forceLoad()

    override fun onCanceled(data: MutableList<T>?) = super.onCanceled(data)

    override fun onStopLoading() {
        cancelLoad()
    }

    override fun onReset() = stopLoading()

}